test/tangle-prune.test.js

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2022 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const test = require('tape')
+const bipf = require('bipf')
+const Testbot = require('./helpers/testbot')
+const tanglePrune = require('../lib/tangle-prune')
+
+const chars = 'abcABC123=+? '.split('')
+const encodedLength = (obj) => bipf.encodingLength(obj)
+const randomChar = () => chars.sort(() => (Math.random() < 0.5 ? -1 : +1))[0]
+const randomText = (length) => {
+  let output = ''
+  while (output.length < length) output += randomChar()
+  return output
+}
+
+test('tangle prune', async (t) => {
+  const ssb = Testbot()
+  const ssbId = ssb.id
+
+  const group = await ssb.tribes2.create()
+
+  const publishSize = async (size, recpCount = 1) => {
+    const content = {
+      type: 'post',
+      text: randomText(size),
+      recps: [group.id, ...new Array(recpCount - 1).fill(ssbId)],
+    }
+
+    return new Promise((resolve, reject) => {
+      ssb.tribes2.publish(content, (err, msg) => {
+        if (err) return resolve(false)
+
+        ssb.db.get(msg.key, (err, msgVal) => {
+          if (err) return reject(err)
+          const plainLength = encodedLength(msgVal.content)
+          resolve(plainLength)
+        })
+      })
+    })
+  }
+
+  async function findMaxSize(numberRecps = 1) {
+    // Apply bisection method to find max size of a message which can be published
+
+    let lower = 4000
+    let mid
+    let upper = 8000
```

Comment:
:fire: set upper to 17000

https://github.com/ssbc/ssb-buttwoo-spec#validation
> the content length in bytes. This number must not exceed 16384.

Reply:
If we change this to 17000 now then later when I do https://github.com/ssbc/ssb-tribes2/issues/10 this test won't start failing. I sort of want it to fail so I know that I actually managed to swap the format. But maybe there are more solid ways to test which format we're using?